modules/programs/firefox.nix

```diff
@@ -663,6 +680,19 @@ in {
           (", namely " + concatStringsSep ", " defaults);
       })
 
+      (let
+        getContainers = profiles:
+          flatten (mapAttrsToList (_: value: value.containers) profiles);
```

Comment:
given that `value.containers` is an attribute set, `mapAttrsToList` is going to return a list of attribute sets. There is nothing to flatten here. For example, the function will probably return something like:

```
[
  { "shopping" = ...; } // 1st profile
  { "shopping" = ...; } // 2nd profile
]
```

I don't think this is what the rest of the code expects.

Reply:
Yeah take a look at the new `getContainers` function, I've also added `attrValues` which then requires `flatten`. Please take an other look.